File: JavaScript/linear_search.js
//function which returns the position of the element from an array
function linearSearch(array, key) {
  const n = array.length;

  // Create a variable found which says whether the key is found or not.
  let found = false;
  let keyIndex = 0;

  // Traverse the array
  for (let i = 0; i < n; ++i) {
    // Check if the i'th element of the array is equal to key(both the value and the type).
    if (array[i] === key) {
      // If yes, make found = True and update the key index and break
      found = true;
      keyIndex = i;
      break;
    }
  }
  // Check whether the key is found or not and return accordingly
  if (found) return keyIndex;
  else return -1;
}

function main() {
  // Linear Search can be applied to any array(both sorted and unsorted)
  const arr = [12, 45, 78, 12, 456, 34, 76, 78, 89, 100];

  // Element to be searched in the array
  const key = 76;

  const pos = linearSearch(arr, key);

  // Check whether the element is present in the array or not with the help of pos variable
  if (pos !== -1) console.log("Element found at position", pos);
  else console.log("Element not found in array.");
}

main();

/*
Input: An array of numbers and the number we are searching
Output: Returns position of the element if it's present if not -1

----------------------------------------
Sample Input:
[12, 45, 78, 12, 456, 34, 76, 78, 89, 100]
key = 76
Output: 
Element found in position 6

Sample Input:
[12, 45, 78, 12, 456, 34, 76, 78, 89, 100]
key = 32
Output:
Element not found in array

----------------------------------------

Time Complexity:
Worst and Average case = O(n)
Best case = O(1)

Space Complexity: O(1)
*/

File: Python/LinearSearch.py
"""
Input : arr[] = [1, 2, 3, 4, 7]
        x = 3;
Output : 2
Element x is present at index 2

Input : arr[] = [1, 2, 3, 4, 7]
           x = 10;
Output : -1
Element x is not present in arr[].

Time Complexity O(n)
Space Complexity O(1)
"""

# Here arr is the array and x is the element to be searched
def linearSearch(arr, x):
    for i in range(len(arr)):
        if arr[i] == x:
            return i    # Return index if found
    return -1   # Return -1 if not found

if __name__=='__main__':
    arr = [1,2,4,5,3,7]
    item1 = 10
    item2 = 4
    print(linearSearch(arr,item1))
    print(linearSearch(arr,item2))
